minitwit/src/main/java/WebApplication.java

```diff
@@ -39,6 +39,34 @@
         public static final String ADD_MESSAGE = "/add_message";
         public static final String FOLLOW = "/:username/follow";
         public static final String UNFOLLOW = "/:username/unfollow";
+
+        /**
+         * Transforms a url with :parameters using a map with keys and values
+         * @param url to use
+         * @param values map of parameters and their values, must be String or Integer values
+         * @return url with parameters replaces with values
+         * @throws Exception
+         */
+        public static String urlFor(String url, Map<String, Object> values) throws Exception {
+            String u = url;
+            for (String key: values.keySet()) {
+                System.out.println(u + " : " + key + " : " + values.get(key));
+                if (values.get(key) instanceof String) {
+                    u = u.replace(":" + key, (String) values.get(key));
+                } else if (values.get(key) instanceof Integer) {
+                    u = u.replace(":" + key, String.valueOf(values.get(key)));
+                } else {
+                    System.out.println("Cant convert value, lets see what happens now... " + key + " : " + values.get(key));
+                    u = u.replace(":" + key, "" + values.get(key));
+                }
+            }
+
+            return u;
+        }
+
+        public static String urlFor(String url) throws Exception {
```

Comment:
What's the point of this method? It doesn't seem to do anything. :P

Reply:
It allows calling urlFor without url parameters like [this](https://github.com/Herover/itu-devops-h/pull/47/files#diff-1b1a0c14b12b0486b3a2b443c33368874d5d5bfe21a01a8752f84153310beae2L32).

Reply:
But at that point, like when you say `urls.urlFor($ADD_MESSAGE)`, is it not the same to simply write `$ADD_MESSAGE`?

Reply:
It is :D